sky/backends/backend_utils.py

```diff
@@ -117,6 +117,9 @@
 _RAY_YAML_KEYS_TO_RESTORE_FOR_BACK_COMPATIBILITY = {
     'cluster_name', 'provider', 'auth', 'node_config'
 }
+_RAY_YAML_KEYS_TO_RESTORE_EXCLUDE_FOR_BACK_COMPATIBILITY = [[
+    'provider', 'region'
+], ['provider', 'availability_zone'], ['provider', 'location']]
```

Comment:
Keep `availability_zone` only, and write a comment on why it's needed? And probably on upper-level code will ensure existing yaml's region field matches the new.

Reply:
Good point! Only keeping the availability_zone now.